draco/analysis/transform.py

```diff
@@ -257,71 +256,81 @@ def process(self, ss):
             if self.weight == "uniform":
                 nprod_in_stack = (nprod_in_stack > 0).astype(np.float32)
 
-        # Find the local times (necessary because nprod_in_stack is not distributed)
-        ntt = ss.vis.local_shape[-1]
-        stt = ss.vis.local_offset[-1]
-        ett = stt + ntt
-
         # Create counter to increment during the stacking.
         # This will be used to normalize at the end.
         counter = np.zeros_like(sp.weight[:])
-
         # Dereference the global slices now, there's a hidden MPI call in the [:] operation.
         spv = sp.vis[:]
         ssv = ss.vis[:]
         spw = sp.weight[:]
         ssw = ss.weight[:]
 
-        # Iterate over products (stacked) in the sidereal stream
-        for ss_pi, ((ii, ij), conj) in enumerate(zip(ss_prod, ss_conj)):
-
-            # Map the feed indices into ones for the Telescope class
-            bi, bj = input_ind[ii], input_ind[ij]
-
-            # If either feed is not in the telescope class, skip it.
-            if bi is None or bj is None:
-                continue
-
-            sp_pi = self.telescope.feedmap[bi, bj]
-            feedconj = self.telescope.feedconj[bi, bj]
-
-            # Skip if product index is not valid
-            if sp_pi < 0:
-                continue
-
-            # Generate weight
-            if self.weight == "inverse_variance":
-                wss = ssw[freq_ind, ss_pi]
-
-            else:
-                wss = (ssw[freq_ind, ss_pi] > 0.0).astype(np.float32)
-                wss.local_array[:] *= nprod_in_stack[np.newaxis, ss_pi, stt:ett]
-
-            # Accumulate visibilities, conjugating if required
-            if feedconj == conj:
-                spv[:, sp_pi] += wss * ssv[freq_ind, ss_pi]
-            else:
-                spv[:, sp_pi] += wss * ssv[freq_ind, ss_pi].conj()
+        # Figure out mapping between the local frequencies
```

Comment:
I tested a daily pipeline run without issue, but I think this will break if the local slice of frequencies in the file and the local slice of frequencies in the telescope instance don't contain all the same frequencies